tools/ci/actions/runner.sh

```diff
@@ -0,0 +1,158 @@
+      --disable-shared \
+      --disable-debug-runtime \
+      --disable-instrumented-runtime \
+      --disable-systhreads \
+      --disable-str-lib \
+      --disable-unix-lib \
+      --disable-bigarray-lib \
+      --disable-ocamldoc \
+      --disable-native-compiler \
+      --enable-ocamltest \
+      --disable-dependency-generation \
+      $CONFIG_ARG"
+  else
+    configure_flags="\
+      --prefix=$PREFIX \
+      --enable-flambda-invariants \
+      --enable-ocamltest \
+      --disable-dependency-generation \
+      $CONFIG_ARG"
+  fi
+  case $XARCH in
+  x64)
+    ./configure $configure_flags
+    ;;
+  i386)
+    ./configure --build=x86_64-pc-linux-gnu --host=i386-linux \
+      CC='gcc -m32' AS='as --32' ASPP='gcc -m32 -c' \
+      PARTIALLD='ld -r -melf_i386' \
+      $configure_flags
+    ;;
+  *)
+    echo unknown arch
+    exit 1
+    ;;
+  esac
+}
+
+Build () {
+  if [ "$MIN_BUILD" = "1" ] ; then
+    if $MAKE world.opt ; then
+      echo "world.opt is not supposed to work!"
+      exit 1
+    else
+      $MAKE world
+    fi
+  else
+    $MAKE world.opt
+    $MAKE ocamlnat
+  fi
+  echo Ensuring that all names are prefixed in the runtime
+  ./tools/check-symbol-names runtime/*.a
+}
+
+Test () {
+  cd testsuite
+  echo Running the testsuite with the normal runtime
+  $MAKE all
+  if [ "$MIN_BUILD" != "1" ] ; then
+    echo Running the testsuite with the debug runtime
+    $MAKE USE_RUNTIME='d' OCAMLTESTDIR="$(pwd)/_ocamltestd" TESTLOG=_logd all
+  fi
+  cd ..
+  if command -v pdflatex &>/dev/null  ; then
+    echo Ensuring that all library documentation compiles
+    $MAKE -C ocamldoc html_doc pdf_doc texi_doc
+  fi
+}
+
+Install () {
+  $MAKE install
+}
+
+Checks () {
+  if fgrep 'SUPPORTS_SHARED_LIBRARIES=true' Makefile.config &>/dev/null ; then
+    echo Check the code examples in the manual
+    $MAKE manual-pregen
+  fi
+  # check_all_arches checks tries to compile all backends in place,
+  # we would need to redo (small parts of) world.opt afterwards to
+  # use the compiler again
+  $MAKE check_all_arches
+  # Ensure that .gitignore is up-to-date - this will fail if any untreacked or
+  # altered files exist.
+  test -z "$(git status --porcelain)"
+  # check that the 'clean' target also works
+  $MAKE clean
+  $MAKE -C manual clean
+  # check that the `distclean` target definitely cleans the tree
+  $MAKE distclean
+  # Check the working tree is clean
+  test -z "$(git status --porcelain)"
+  # Check that there are no ignored files
+  test -z "$(git ls-files --others -i --exclude-standard)"
+}
+
+CheckManual () {
+      cat<<EOF
+--------------------------------------------------------------------------
+This test checks the global structure of the reference manual
+(e.g. missing chapters).
+--------------------------------------------------------------------------
+EOF
+  # we need some of the configuration data provided by configure
+  ./configure
+  $MAKE check-stdlib check-case-collision -C manual/tests
+
+}
```

Comment:
This function is not called anywhere.  Do you intend to do something with it, or is it to be removed eventually?

Also I find it strange that it re-runs configure.  Is the caller not supposed to have called  Configure before?

Reply:
I started by copying the Travis script and deleting everything which was Travis-specific. I left this one here because it looked like we would want to move the manual build off Travis too at some point, but I didn't want to test it at this stage. Indeed, the pipeline wouldn't need to run configure again.